src/ansys/simai/core/data/models.py

```diff
@@ -99,6 +100,13 @@ def build(
                 new_model = simai.models.build(build_conf)
 
         """
+        if not configuration.project:
+            raise InvalidArguments("The model configuration does not have a project set")
+
+        if not configuration.project.is_trainable:
+            raise InvalidArguments(
+                f"Cannot train model because: {configuration.project.is_trainable.reason}"
+            )
```

Comment:
We're doing 2 API calls instead of one here (calling `is_trainable` twice), one of the many reasons I hate hidden API calls